8.frontend-react/crm-frontend/src/components/miniProjectsList/MiniProjectsList.jsx

```diff
@@ -0,0 +1,118 @@
+import React, { useEffect, useState, useRef } from 'react';
+import './miniProjectsList.scss';
+import Table from '../table/Table';
+import Modal from 'react-modal';
+import Form from '../form/Form';
+import CrmApi from '../../helpers/CrmApi';
+import statusMap from '../../helpers/StatusMap';
+import { useHistory } from 'react-router';
+
+const crmApi = new CrmApi();
+
+function MiniProjectsList(props){
+    const [projects, setProjects] = useState([]);
+    const [isProjectModalOpen, setIsProjectModalOpen] = useState(false);
+    const [modalProjectDetails, setModalProjectDetails] = useState({});
+    const [projectDetails, setProjectDetails] = useState({});
+    const projectDetailsRef = useRef(projectDetails);
+    projectDetailsRef.current = projectDetails;
+  
+    const history = useHistory();
+
+    
+    useEffect(()=>{
+        setProjects(props.projectsList);
+    }, [props.projectsList])
+
+
+    const submitUpdateProject = async (dataToSent) => {
+      const res = await crmApi.postRequest("/projects/updateProject/", {projectId: projectDetailsRef.current.project_id, user: true, set:{project_status: statusMap.inProgress.key, estimated_time: dataToSent.hours.value}});
```

Comment:
extract the data sent to the api to a variable , 
to make it more readable . 
dont be afraid of new lines 😉 
```suggestion
      const data = {
           projectId: projectDetailsRef.current.project_id, 
           user: true, 
           set:{
                project_status: statusMap.inProgress.key, 
                estimated_time: dataToSent.hours.value
                }
          }
      const res = await crmApi.postRequest("/projects/updateProject/", data);
```